compiler/hash-semantics/src/new/environment/tc_env.rs

```diff
@@ -62,14 +70,47 @@ macro_rules! tc_env {
 
 pub type DiagnosticsStore = DiagnosticCellStore<SemanticError, SemanticWarning>;
 
+pub type PreludeOrUnset = OnceCell<ModDefId>;
+
 tc_env! {
     #hide env: Env<'tc>,
     diagnostics: DiagnosticsStore,
     ast_info: AstInfo,
+    prelude_or_unset: PreludeOrUnset,
     primitives_or_unset: DefinedPrimitivesOrUnset,
     intrinsics_or_unset: DefinedIntrinsicsOrUnset,
 }
 
+pub struct SemanticStorage {
```

Comment:
This should have the `EntryPointState` in some state for the codegen and later.

Reply:
Yes, I will add this in next PR